modules/layers/src/geojson-layer/geojson-layer.js

```diff
@@ -19,53 +19,43 @@
 // THE SOFTWARE.
 
 import {CompositeLayer, log} from '@deck.gl/core';
-import ScatterplotLayer from '../scatterplot-layer/scatterplot-layer';
-import PathLayer from '../path-layer/path-layer';
-// Use primitive layer to avoid "Composite Composite" layers for now
-import SolidPolygonLayer from '../solid-polygon-layer/solid-polygon-layer';
 import {replaceInRange} from '../utils';
 import {binaryToFeatureForAccesor} from './geojson-binary';
+import {
+  POINT_LAYER,
+  LINE_LAYER,
+  POLYGON_LAYER,
+  getDefaultProps,
+  forwardProps
+} from './sub-layer-map';
 
 import {getGeojsonFeatures, separateGeojsonFeatures} from './geojson';
 import {createLayerPropsFromFeatures, createLayerPropsFromBinary} from './geojson-layer-props';
 
-const defaultLineColor = [0, 0, 0, 255];
-const defaultFillColor = [0, 0, 0, 255];
-
 const defaultProps = {
+  ...getDefaultProps(POINT_LAYER.circle),
+  ...getDefaultProps(POINT_LAYER.icon),
+  ...getDefaultProps(POINT_LAYER.text),
+  ...getDefaultProps(LINE_LAYER),
+  ...getDefaultProps(POLYGON_LAYER),
+
+  // Overwrite sub layer defaults
   stroked: true,
   filled: true,
   extruded: false,
   wireframe: false,
+  iconAtlas: {type: 'object', value: null},
```

Comment:
Why do you need to redefine some properties but you don't for others?

For example, you redefine `getIcon`, but you don't redefine `getFillColor`. Is this because getFillColor is already defined at `SolidPolygonLayer`?

Reply:
`getIcon`'s default is `d => d.icon` which definitely does not apply to GeoJSON.

`getFillColor`'s default is `[0,0,0,255]` which works fine.

`iconAtlas` and `iconMapping` are async props which cannot be directly forwarded.